### ISSUE TITLE
Incompatible services.phpfpm change

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#17253]
## Issue description
PR https://github.com/NixOS/nixpkgs/pull/16132 removed ```services.phpfpm.poolConfigs``` and replaced
it with ```services.phpfpm.pools``` of different option type. Before the NixOS 16.09 release I think we should either revert that change or bring back the  ```services.phpfpm.poolConfigs``` interface, but print warnings when being used.

**Comment by ericsagnes** [comment]
Maybe we should introduce a `mkChangedOptionModule` for such cases that would print something like:

```
The option definition X.Y.Z in FILE has been changed to a A.B.C that has a different option type. Update the configuration accordingly to A.B.C documentation.
```

**Comment by bjornfor** [comment]
If I understand that proposal correctly, I don't think it's sufficient. I think a configuration.nix file should not break between release N and release N+1. IMHO, that message above should just be a warning and the old interface should be working until release N+2.

Will bringing back the old interface and adding a "trace" message, if being used, do the trick?

**Comment by ericsagnes** [comment]
> If I understand that proposal correctly, I don't think it's sufficient. I think a configuration.nix file should not break between release N and release N+1.

Right, I don't know the policy for option changes but AFAIK there are a few documented incompatible changes coming with each [release](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-1603.xml#L119).

While modules are moving fast it might be difficult, but this could be a good occasion to set guidelines regarding incompatible changes in options.

**Comment by bjornfor** [comment]
Yes, breaking has happened before and will happen in the future. I just want to minimize them.

I've opened a PR to revert the phpfpm interface change: https://github.com/NixOS/nixpkgs/pull/17285

**Comment by bjornfor** [comment]
Fixed: c7860cae1a4b54f4b54339142472f00f861fa055.
